logstash-core/lib/logstash/instrument/metric_store.rb

```diff
@@ -180,6 +180,16 @@ def each(path = nil, &block)
     end
     alias_method :all, :each
 
+    def prune(path)
+      key_paths = path.gsub(/^#{KEY_PATH_SEPARATOR}+/, "").split(KEY_PATH_SEPARATOR)
```

Comment:
Maybe a good time to create a `#key_paths(path)` method

We have the same code here https://github.com/elastic/logstash/pull/5819/files#diff-ae63e631134ccf36b5e761ae91487f61R83